optuna/study.py

```diff
@@ -466,6 +467,35 @@ def _create_record_and_aggregate_column(trial):
 
         return df
 
+    def enqueue_trial(
+            self,
+            params=None,  # type: Optional[Dict[str, Any]]
+            user_attrs=None,  # type: Optional[Dict[str, Any]]
+            system_attrs=None  # type: Optional[Dict[str, Any]]
+    ):
```

Comment:
Just dropping a though/leaving a note here but it might make sense to also accept `FixedTrial` objects as arguments as the idea is quite similar, to fix parameters.

Reply:
That makes sense. However, I'm not sure whether allowing `FixedTrial` is actually beneficial to users because there is no handy way to build `FixedTrial`, e.g., from existing trials.

Reply:
This is not a strong opinion but I was thinking about the signature, namely if we cannot omit "_trial" from "enqueue_trial" to simplify the name, and more importantly (also?) accept `FixedTrial` and/or `FrozenTrial`, or sequences of those, i.e. `Study.enque(trials: Union[FixedTrial, FrozenTrial, Sequence[Union[FixedTrial, FrozenTrial]])`. I guess this depends on how it's going to be used and I might not be best person to answer.

Reply:
I see. Thank you for your explanation.
The following comment is just a random thought.

---

I agree that if this method takes `FixedTrial` as the argument, we can omit `_trial` suffix from the name, and it would be useful, for some users, to allow passing a list of `FixedTrial`.
On the other hand, I think that it would be better not to accept `FrozenTrial` as the argument since the class is also used to represent finished trials and contains a lot of noisy fields (e.g., value, number, state, distributions) that aren't relevant to the queueing feature. Instead, I would like to recommend to add a method like `to_fixed_trial()` to `FrozenTrial`.

Reply:
It sounds both opinions are reasonable. From the method name of `enqueue_trial`, most users imagine that its argument is a trial object.

It sounds passing `FixedTrial` is relatively good, but it's not originally designed for this purpose. I personally feel that current alternative solutions are not perfect. I relatively prefer just renaming the method (e.g. `study.enqueue_next_search_params(params)`).

It's an important and difficult discussion. It might be better to delay the release version.

Reply:
>  It's an important and difficult discussion. I'm OK to delay the release version.

Agree.

Reply:
We discussed on a video chat.

* For now, `enqueue_trial` accepts `param: Dict[str, Any]` only. It might become to accept Trial-like objects (e.g. FrozenTrial) in the future.
    * rev: fe4f09e77482
* Mark `enqueue_trial` as an experimental feature.
    * rev: 6163b95